File: SitefinityWebApp/Modules/Testimonials/ControlDesigners/SubmitTestimonialDesigner.js
Type.registerNamespace("SitefinityWebApp.Modules.Testimonials.ControlDesigners");

SitefinityWebApp.Modules.Testimonials.ControlDesigners.SubmitTestimonialDesigner = function (element) {
    SitefinityWebApp.Modules.Testimonials.ControlDesigners.SubmitTestimonialDesigner.initializeBase(this, [element]);
}


SitefinityWebApp.Modules.Testimonials.ControlDesigners.SubmitTestimonialDesigner.prototype = {
    initialize: function () {
        SitefinityWebApp.Modules.Testimonials.ControlDesigners.SubmitTestimonialDesigner.callBaseMethod(this, 'initialize');

    },
    dispose: function () {
        SitefinityWebApp.Modules.Testimonials.ControlDesigners.SubmitTestimonialDesigner.callBaseMethod(this, 'dispose');
    },
    refreshUI: function () {
        var data = this._propertyEditor.get_control();
        jQuery("#AutoPublish").attr('checked', data.AutoPublish);
    },
    applyChanges: function () {

        var controlData = this._propertyEditor.get_control();

        controlData.AutoPublish = jQuery("#AutoPublish").attr('checked') == "checked";
    }
}

SitefinityWebApp.Modules.Testimonials.ControlDesigners.SubmitTestimonialDesigner.registerClass('SitefinityWebApp.Modules.Testimonials.ControlDesigners.SubmitTestimonialDesigner', Telerik.Sitefinity.Web.UI.ControlDesign.ControlDesignerBase);
if (typeof (Sys) !== 'undefined') Sys.Application.notifyScriptLoaded();

File: SitefinityWebApp/Modules/Testimonials/ControlDesigners/SubmitTestimonialDesigner.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Telerik.Sitefinity.Web.UI.ControlDesign;
using System.Web.UI;

namespace SitefinityWebApp.Modules.Testimonials.ControlDesigners
{
	public class SubmitTestimonialDesigner : ControlDesignerBase
	{
		/// <summary>
		/// Initializes the controls.
		/// </summary>
		/// <param name="container">The container.</param>
		protected override void InitializeControls(Telerik.Sitefinity.Web.UI.GenericContainer container)
		{
			base.DesignerMode = ControlDesignerModes.Simple;
		}

		private string _layoutTemplatePath = "~/Modules/Testimonials/ControlDesigners/SubmitTestimonialDesignerTemplate.ascx";
		/// <summary>
		/// Gets or sets the layout template path.
		/// </summary>
		/// <value>
		/// The layout template path.
		/// </value>
		public override string LayoutTemplatePath
		{
			get { return _layoutTemplatePath; }
			set { _layoutTemplatePath = value; }
		}

		private string _scriptPath = "~/Modules/Testimonials/ControlDesigners/SubmitTestimonialDesigner.js";
		/// <summary>
		/// Gets or sets the designer script path.
		/// </summary>
		/// <value>
		/// The designer script path.
		/// </value>
		public string DesignerScriptPath
		{
			get { return _scriptPath; }
			set { _scriptPath = value; }
		}

		/// <summary>
		/// Gets the name of the layout template.
		/// </summary>
		/// <value>
		/// The name of the layout template.
		/// </value>
		protected override string LayoutTemplateName
		{
			get { return null; }
		}

		/// <summary>
		/// Gets the script references.
		/// </summary>
		/// <returns></returns>
		public override IEnumerable<ScriptReference> GetScriptReferences()
		{
			var scripts = base.GetScriptReferences() as List<ScriptReference>;
			if (scripts == null) return base.GetScriptReferences();

			scripts.Add(new ScriptReference(DesignerScriptPath));
			return scripts.ToArray();
		}
	}
}